src/Analyzers/Analyzers/test/StartupFactsTest.cs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Analyzers.TestFiles.StartupFactsTest;
+using Microsoft.CodeAnalysis;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Analyzers
+{
+    public class StartupFactsTest : AnalyzerTestBase
+    {
+        [Theory]
```

Comment:
😞

Reply:
Yeah, it's really the most reasonable thing to do for these cases.